### PR TITLE
feat: add options to boa

### PR DESCRIPTION
**Pull request by elewis787**
Working on refactoring the boa options pattern

### CODE REVIEW

**Comment by elewis787** [comment]
closes #4

**Comment by elewis787** [comment]
@jeff1010322 - ready for you to take a look - This got a little bigger than I wanted but the gist is - I introduced a new object to build the options off of. This is to plan for future configurations and overall clean up the use of package level variables. 

I would love your thoughts on if the styles should be public or set through options as well

File: options.go [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		opt.width = w
+	})
+}
+
+func WithAltScreen(b bool) Options {
+	return newFuncOption(func(opt *options) {
+		if !b {
+			opt.atlScreen = noOpt
+		}
+		opt.atlScreen = tea.WithAltScreen()
```
Comment (jeff1010322) [inline options.go]: Doesn't this always overwrite opt.altScreen with the `tea.WithAltScreen()`?

File: options.go [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package boa
+
+import tea "github.com/charmbracelet/bubbletea"
+
+type options struct {
+	// public
+	atlScreen  tea.ProgramOption
```
Comment (jeff1010322) [inline options.go]: Looks like this should be `altScreen` t and l are swapped.

File: options.go (hunk 1 above)
Comment (elewis787) [reply options.go]: fixed

File: options.go (hunk 2 above)
Comment (elewis787) [reply options.go]: fixed

**Review by jeff1010322 — COMMENTED** [review]
Overall I like the options, but I think it would be nice if the styles would be added as options as well instead of using the Public styles option to update them.

**Comment by elewis787** [comment]
> 

I agree - I will switch them over!

**Comment by elewis787** [comment]
@jeff1010322 - please take another look

**Review by jeff1010322 — APPROVED** [review]
LGTM!

**Pull request merged**